spec/cirro_io/client/base_spec.rb

```diff
@@ -38,4 +45,27 @@
       described_class.custom_post('bulk/custom-endpoint', { a: :b })
     end
   end
+
+  describe 'configuration' do
+    let(:other_site)    { 'https://api.other.cirro.io' }
+    let(:other_version) { 'vXXX' }
+
+    before do
+      CirroIO::Client::AppWorker.site = "#{other_site}/#{other_version}"
+    end
+
+    it 'supports multiple backends' do
+      stub_request(:get, "#{other_site}/#{other_version}/app-workers/1")
+        .with(headers: request_headers)
+        .to_return(body: File.read('spec/fixtures/app_worker.json'), headers: response_headers)
+
+      expect(CirroIO::Client::AppWorker.find(1).first.id).to eq('1')
+
+      stub_request(:get, "#{test_site}/v1/app-users/3")
```

Comment:
I confess I am a bit confused. It seems like `site` has effectively become an array you can push values into and then you can make calls to either site. But I guess I didn't realize that the API calls included the base site in the call themselves. But is my mental model correct?

Reply:
I'm not sure, I understand what do you mean 🤔

In fact, every `JsonApiClient:: Resource` has a `site`. Though it's inheritable by design, changing it doesn't work out of the box due to a design error in `JsonApiClient`. This very error is handled by the hack in this PR.

`CirroIO::Client.configuration` sets `CirroIO::Client::Base.site`. It can be overriden in descendant classes. Thus, every resource class has _one_ `site` at a time. There are no collections of them or any connection pulls.

Reply:
I suppose how we gonna use it is in the initializer we do:
```
require 'cirro_io/client'

private_key = Rails.env.production? ? '/var/secrets/cirroio-private-key/key.json' : './cirro_dev_private_key.pem'

CirroIO::Client.configure do |c|
  c.app_id Settings.cirro.app_id
  c.private_key_path private_key
  c.site Settings.cirro.site
end

CirroIO::Client::Gig.site = Setting.shadow_cirro.site
CirroIO::Client::GigTask.site = Setting.shadow_cirro.site
CirroIO::Client::GigResult.site = Setting.shadow_cirro.site
...
```
Would this work?